core/src/main/java/org/axonframework/queryhandling/GenericSubscriptionQueryUpdateMessage.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2010-2018. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.queryhandling;
+
+import org.axonframework.messaging.GenericMessage;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.MessageDecorator;
+import org.axonframework.messaging.MetaData;
+
+import java.util.Map;
+
+/**
+ * Generic {@link SubscriptionQueryUpdateMessage} which holds incremental update of an subscription query.
+ *
+ * @param <U> type of incremental update
+ * @author Milan Savic
+ * @since 3.3
+ */
+public class GenericSubscriptionQueryUpdateMessage<U> extends MessageDecorator<U>
+        implements SubscriptionQueryUpdateMessage<U> {
+
+    private static final long serialVersionUID = 5872479410321475147L;
+
+    /**
+     * Creates {@link GenericSubscriptionQueryUpdateMessage} from provided {@code payload} which represents incremental
+     * update. The provided {@code payload} may not be {@code null}.
+     *
+     * @param payload incremental update
+     * @param <T>     type of the {@link GenericSubscriptionQueryUpdateMessage}
+     * @return created message
```

Comment:
I'd probably rephrase this to `created a {@link SubscriptionQueryUpdateMessage} with the given {@code payload}.`